packages/core/src/controllers/NetworkController.ts

```diff
@@ -10,6 +10,8 @@ export interface NetworkControllerClient {
     approvedCaipNetworkIds: NetworkControllerState['approvedCaipNetworkIds']
     supportsAllNetworks: NetworkControllerState['supportsAllNetworks']
   }>
+  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents, @typescript-eslint/no-explicit-any
+  getWalletConnectProvider?: () => Promise<any | undefined>
```

Comment:
This looks like a pattern going against what we have established via connectors in core. What is this, why is it getting provider in network controller? Can we use existing walletConnect connector pattern?